supervision/tracker/byte_tracker/core.py

```diff
@@ -60,7 +62,7 @@ def activate(self, kalman_filter, frame_id):
 
         self.tracklet_len = 0
         self.state = TrackState.Tracked
-        if frame_id == 1:
+        if frame_id == 1 and self.minimum_consecutive_frames == 1:
```

Comment:
We could solve the problem of cycling through tracker IDs too quickly if we actually assigned a tracker_id only when that condition is met.

Reply:
Hmm. This condition only handles the first frame of the video. I considered only assigning a `tracker_id` once the condition on line 98 was met, (track has existed for longer than `minimum_consecutive_frames`), but that would break the rest of the tracker because it relies on tracker_id to identify different tracks in joint_stracks and sub_stracks, so tracker_id must have a unique value as soon as it gets activated.

One possible solution is to create an additional variable in the STrack object called something like `internal_tracker_id` that is used for all of the ByteTrack internal logic, and then leave the original `tracker_id` variable to only get assigned when a track has existed for long enough. This would allow the tracker to function as usual, and would make the output detections have reasonable tracker_ids.

I also considered just using the `tracker_id` variable and when a track is first initialized assigning the `tracker_id` to a temporary value with a new function just like `next_id` called `next_temporary_id()`. Then once a track has existed for `minimum_consecutive_frames`, the tracker_id would be reassigned to a valid track_id with `next_id()`. But I believe this solution would be problematic because there is a high likelihood that the `next_temporary_id()` and `next_id()` values would overlap, causing two tracks to share the same tracker_id. This could be solved by making the temporary_id's negative, but I don't know if we want to do that.

The final solution I have thought of is redesigning sub_stracks and joint_stracks to use iou overlap like remove_duplicate_tracks does instead of relying on tracker_id to identify tracks, which would allow more freedom to use tracker_id in other ways. (this would slow down the tracker a bit because of the linear assignment problem)

Reply:
Hi @rolson24 👋🏻 

I'm very sorry for the lack of response. I somehow missed the email notification that you had replied.

In that case, I am leaning towards a solution that introduces an internal and external tracker ID. The internal one can have arbitrary values, while the external one should have consecutive tracker IDs.

Reply:
Ok I have implemented this and it seems to be working. 
Here you can see the end of the video with `minimum_consecutive_frames = 1`. The tracker ids have gotten to quite a high number for a 13 second video (463!)
![add_min_frames_test_video_1_frame](https://github.com/roboflow/supervision/assets/99894460/d06cf9ad-ad00-4c28-9be3-f67298606466)

And here is the video for `minimum_consecutive_frames = 1`:

https://github.com/roboflow/supervision/assets/99894460/a2a32b09-c069-4b11-9767-094626067ac2


And here you can see the same area of the video with `minimum_consecutive_frames = 10`. The tracker ids have now only reached 110, and the person in the bottom middle is not tracked because the track hasn't existed for long enough.
![add_min_frames_test_video_15_frames](https://github.com/roboflow/supervision/assets/99894460/a1443075-18b7-41a9-8bb0-4aaf2e8803d2)

Here is the video for `minimum_consecutive_frames = 10`:

https://github.com/roboflow/supervision/assets/99894460/c2f24125-bbbf-437b-bb4e-d9efe6fb7054

Reply:
Hi @rolson24 👋🏻 `#463` seems high. It is pretty much the same result we got here: https://github.com/roboflow/supervision/pull/1076#issuecomment-2039904362

Reply:
Hmm i just checked with the release version and you seem to be correct. I will take a look.